lld/ELF/ScriptParser.cpp

```diff
@@ -833,7 +833,8 @@ constexpr std::pair<const char *, unsigned> typeMap[] = {
 // "(TYPE=<value>)".
 // Tok1 and Tok2 are next 2 tokens peeked. See comment for
 // readSectionAddressType below.
-bool ScriptParser::readSectionDirective(OutputSection *cmd, StringRef tok1, StringRef tok2) {
+bool ScriptParser::readSectionDirective(OutputSection *cmd, StringRef tok1,
+                                        StringRef tok2) {
```

Comment:
This particular case isn't illustrative of it, but I think it is worth considering using some of the "if it doesn't fit on one line, give each element its own line" options in the format. These are at least:

- [BinPackArguments](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#binpackarguments)
- [BinPackParameters](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#binpackparameters)
- [PackConstructorInitializers](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#packconstructorinitializers)
- Several options following the naming pattern [AllowShortXOnASingleLine](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#allowshortblocksonasingleline)

The value is particularly visible when there are  10+ parameters or member initializers: it is easy to lose track of some bin-packed elements, and there is some "write-amplification" which balloons the diff when the bin-packing reflows every line because an element in the middle is changed, added, or removed.

Reply:
There is a big danger though that going too far of the current settings will put a lof of unnecessary churn on existing code (that was mostly formatted with the LLVM defaults). (At least I think it could be a lot more churn than the `AllowShortFunctionsOnASingleLine` I mentioned above...)

Reply:
Makes sense, and I don't want to bog down the effort so I'm happy to just drop this. If I find it compelling enough I can always try to fight for it after we get everything `clang-format` clean